### PR TITLE
Add build targets for Java lite runtime.

### PR DESCRIPTION
**Pull request by xfxyjwf**
Fixes https://github.com/google/protobuf/issues/2762

### CODE REVIEW

**Comment by xfxyjwf** [comment]
@cgrushko

File: BUILD [hunk 1]
```diff
@@ -539,13 +539,103 @@ internal_gen_well_known_protos_java(
     srcs = WELL_KNOWN_PROTOS,
 )
 
+JAVA_LITE_FILES = [
```
Comment (cgrushko) [inline BUILD]: Why not inline this into the rule itself?
There's an internal doc about why this is harmful, I can send you a link.

Comment (xfxyjwf) [inline BUILD]: Done.

File: BUILD [hunk 2]
```diff
@@ -540,12 +540,100 @@ internal_gen_well_known_protos_java(
...
+        "java/core/src/main/java/com/google/protobuf/Service.java",
+        "java/core/src/main/java/com/google/protobuf/ServiceException.java",
+        "java/core/src/main/java/com/google/protobuf/SingleFieldBuilder.java",
+        "java/core/src/main/java/com/google/protobuf/SingleFieldBuilderV3.java",
+        "java/core/src/main/java/com/google/protobuf/TextFormat.java",
+        "java/core/src/main/java/com/google/protobuf/TextFormatParseInfoTree.java",
+        "java/core/src/main/java/com/google/protobuf/TextFormatParseLocation.java",
+        "java/core/src/main/java/com/google/protobuf/UnknownFieldSet.java",
+        "java/core/src/main/java/com/google/protobuf/UnsafeByteOperations.java",
+    ] + [
```
Comment (cgrushko) [inline BUILD]: Delete "] + [" ?

File: BUILD [hunk 3]
```diff
@@ -540,12 +540,100 @@ internal_gen_well_known_protos_java(
 )
 
 java_library(
+    name = "protobuf_java_lite",
```
Comment (cgrushko) [inline BUILD]: if the entire code is in `java/core/src/main/java/com/google/protobuf/`, perhaps put a BUILD file there?

File: BUILD [hunk 4]
```diff
@@ -540,12 +540,100 @@ internal_gen_well_known_protos_java(
...
+cc_binary(
+    name = "protoc_gen_javalite",
+    srcs = ["src/google/protobuf/compiler/java/java_lite_main.cc"],
+    linkopts = LINK_OPTS,
+    visibility = ["//visibility:public"],
+    deps = [":protoc_lib"],
+)
+
+java_library(
     name = "protobuf_java",
```
Comment (cgrushko) [inline BUILD]: Why is this needed, btw? if we're in the branch that only does javalite.

File: BUILD (hunk 2 above)
Comment (xfxyjwf) [inline BUILD]: Done.

File: BUILD (hunk 4 above)
Comment (xfxyjwf) [inline BUILD]: This branch is forked from master branch, so there are all sorts of non-lite stuff here (e.g., all the full Java runtime source code + all other languages are still in this javalite branch). I would rather keep these as-is.

File: BUILD (hunk 3 above)
Comment (xfxyjwf) [inline BUILD]: "java/core/src/main/java/com/google/protobuf" is a bad build package containing completely unrelated stuff (the path is specially crafted for Maven). Putting the target in the root BUILD file is much more cleaner.

**Comment by xfxyjwf** [comment]
No, I don't think cl.bazel.io covers javalite. We are setting up kokoro tests to replace travis/jenkins and bazel tests can be setup as part of it.
